### ISSUE TITLE
Réinstallation de occtax après délocalisation des répertoires

### ISSUE THREAD

**Issue by patricklbs** [PnX-SI/GeoNature#750]
Bonjour,

J'ai essayé de délocaliser les application geonature, taxhub et usershub. J'ai réinstallé tout je ne comprends pas pourquoi le module occtax ne se réinstalle pas.

si je réinstalle le module j'ai cette erreur 

geonature install_gn_module /appli/vgn/geonature/contrib/occtax /occtax --build=false

/appli/vgn/geonature/backend/venv/lib/python3.6/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)
checking file
...ok

checking manifest
...ok
Error while installing GN module 'OCCTAX'.The process returned:
        The module OCCTAX is already installed, but maybe not activated

Une idée ? Lors d'une délocalisation des application y a t il une manip spécifique à faire pour retrouver les bon chemins hormis la réinstall ?

Merci !!

**Comment by jbrieuclp** [comment]
Supprimer la ligne du module dans la table gn_commons.t_modules ?

**Comment by gildeluermoz** [comment]
non surtout pas. Le id_module est utilisé à plusieurs endroits, notamment en synthèse mais aussi dans la gestion des permissions.
Il faudrait plutôt checker s'il existe et recycler l'id existant.

**Comment by jbrieuclp** [comment]
A la réinstallation la ligne sera de nouveau ajoutée dans la BD, avec par défaut sont ID OCCTAX, non ?

**Comment by patricklbs** [comment]
Donc ça veut dire que je dois tout réinstaller les schémas et refaire l'install ?

**Comment by jbrieuclp** [comment]
Non, moi je parlais juste de supprimer la ligne correspondant au module OCCTAX qui est présente dans la table gn_commons.t_modules avant de lancer la commande install_gn_module.

Mais @gildeluermoz indique qu'il ne faut pas, alors bon...

**Comment by DonovanMaillard** [comment]
Si j'ai bien compris, il est possible de supprimer la ligne, mais en remettant /recycler son ID original par la suite. Ca permet de ne pas poser de soucis dans la synthèse (t sources) le cruved etc

**Comment by TheoLechemia** [comment]
S'il il y a déjà des données Occtax, des permissions définit, effectivement, ce n'est pas une bonne idée de supprimer la ligne. De toute façon on ne pourra pas car il y a des clés étrangères.
Si occtax est vide, il n'y a pas de problème.
Pourquoi réinstaller le module si vous avez simplement déplacé vos répertoires ? Quels sont les erreurs remontés ?

**Comment by patricklbs** [comment]
> 
> 
> S'il il y a déjà des données Occtax, des permissions définit, effectivement, ce n'est pas une bonne idée de supprimer la ligne. De toute façon on ne pourra pas car il y a des clés étrangères.
> Si occtax est vide, il n'y a pas de problème.
> Pourquoi réinstaller le module si vous avez simplement déplacé vos répertoires ? Quels sont les erreurs remontés ?

Car il y a des adherences par rapport aux chemins d'install comme un lien symbolique par exemple entre /contrib/occtax et external_module entre autre du coup ça ne fonctionne pas très bien.

En cas de de déplacement de répertoires vous préconisez quels manips ? Comme j'ai des sauvegardes je peux refaire la manip.
Merci

**Comment by TheoLechemia** [comment]
Est-ce qu'il y a des données dans le module ?
Si non supprimer la ligne comme l'a dit @jbrieuclp  et réinstaller le module.
Si oui, recréer le lien symbolique cassé entre le repertoire "external_modules" et le répertoire "contrib/occtax" ou se trouve initialement le module.

**Comment by jbrieuclp** [comment]
Sinon pour fourbe, nommer en OCCTAX_A_CONSERVER le module OCCTAX de la table gn_commons.t_module.
Faire l'installation du module en OCCTAX, supprimer la ligne nouvellement créée sous l'ID OCCTAX dans gn_commons.t_module 
Renommer OCCTAX_A_CONSERVER en OCCTAX 
C'est complètement à la ni vu ni connu je t'embrouille, mais ça devrait fonctionner !

**Comment by patricklbs** [comment]
J'ai recréé les deux liens symboliques lorsque je veux accéder au module occtax je n'ai rien 
![image](https://user-images.githubusercontent.com/1314444/67016727-f4bb9300-f0f8-11e9-9fae-9418235bd502.png)

**Comment by patricklbs** [comment]
> 
> 
> Sinon pour fourbe, nommer en OCCTAX_A_CONSERVER le module OCCTAX de la table gn_commons.t_module.
> Faire l'installation du module en OCCTAX, supprimer la ligne nouvellement créée sous l'ID OCCTAX dans gn_commons.t_module
> Renommer OCCTAX_A_CONSERVER en OCCTAX
> C'est complètement à la ni vu ni connu je t'embrouille, mais ça devrait fonctionner !

:) je vais essayer

**Comment by patricklbs** [comment]
Ah non c'est bon il faut cliquer sur explorer la synthèse.

**Comment by patricklbs** [comment]
> 
> 
> Sinon pour fourbe, nommer en OCCTAX_A_CONSERVER le module OCCTAX de la table gn_commons.t_module.
> Faire l'installation du module en OCCTAX, supprimer la ligne nouvellement créée sous l'ID OCCTAX dans gn_commons.t_module
> Renommer OCCTAX_A_CONSERVER en OCCTAX
> C'est complètement à la ni vu ni connu je t'embrouille, mais ça devrait fonctionner !

Ca marche !! ;) Merci.
